cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,243 @@
+"""
+Tests for the maintenance app views.
+"""
+import ddt
+
+from django.core.urlresolvers import reverse
+from student.tests.factories import AdminFactory, UserFactory
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from contentstore.management.commands.utils import get_course_versions
+
+from .views import COURSE_KEY_ERROR_MESSAGES, get_maintenace_urls
+
+
+class TestMaintenanceIndex(SharedModuleStoreTestCase):
+    """
+    Tests for maintenance index view.
+    """
+
+    def setUp(self):
+        """Make the user global staff. """
+        super(TestMaintenanceIndex, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+        self.view_url = reverse('maintenance:maintenance')
+
+    def test_maintenance_index(self):
+        response = self.client.get(self.view_url)
+        self.assertContains(response, "Maintenance", status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in get_maintenace_urls():
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class MaintenanceViewTestCase(SharedModuleStoreTestCase):
```

Comment:
@adampalay Isn't publishing the course action of modifying it ? So why do we want to use `SharedModuleStoreTestCase` which should only be used if individual tests do not modify the course. As described in [SharedModuleStoreTestCase](https://github.com/edx/edx-platform/pull/9070/files#diff-480f335443b1a6487aeb0282e7d6cd31R222) class which states :

> Use this class if your tests set up one or a small number of courses that individual tests do not modify. If your tests modify contents in the ModuleStore, you should use ModuleStoreTestCase instead.